### PR TITLE
docs: update the no-required-schema-properties-undefined rule description

### PR DESCRIPTION
**Pull request by tatomyr**
## What/Why/How?

It might be unclear that the rule is case sensitive. Added an example to illustrate that.

## Reference

See the comment: https://github.com/Redocly/redocly-cli/issues/477#issuecomment-1943467337

## Has code been changed?

- [ ] Tested with redoc/reference-docs/workflows (internal)
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by tatomyr** [comment]
@lornajane do you recall any recent changes to the vale linting process?

**Comment by lornajane** [comment]
Not very recently - #1390 but it was still correctly failing (if that makes sense!) after that merged.

File: docs/rules/no-required-schema-properties-undefined.md
Comment (lornajane) [inline docs/rules/no-required-schema-properties-undefined.md:88]:
```suggestion
The rule is case-sensitive, which means a property `name` does not match the string `Name` in the `required` list:
```

**Comment by lornajane** [comment]
Let's definitely not hold up this PR for Vale, I'll pick it up separately

**Review by lornajane — APPROVED** [review]
Much clearer, thanks!

**Pull request merged**
